onnx/common/ir_pb_converter.cc

```diff
@@ -190,13 +190,13 @@ std::vector<Dimension> tensorShapeProtoToDimensions(const ONNX_NAMESPACE::Tensor
   dims.reserve(tsp.dim_size());
   for (int i = 0; i < tsp.dim_size(); i++) {
     if (tsp.dim(i).has_dim_value()) {
-      dims.push_back(Dimension(static_cast<int>(tsp.dim(i).dim_value())));
+      dims.emplace_back(static_cast<int>(tsp.dim(i).dim_value()));
     } else if (tsp.dim(i).has_dim_param()) {
-      dims.push_back(Dimension(tsp.dim(i).dim_param()));
+      dims.emplace_back(tsp.dim(i).dim_param());
     } else {
       // a dimension that has neither dim_value nor dim_param set
       // represents an unknown dimension unrelated to other unknown dimensions.
-      dims.push_back(Dimension());
+      dims.emplace_back();
```

Comment:
Performance improvement: `emplace_back(...)` is more effective than `push_back(Dimension(...))` because it constructs the `Dimension` object in-place.